src/parse-mockdb.js

```diff
@@ -101,9 +101,24 @@ function runHook(className, hookType, data) {
     const modelData = Object.assign(new Object, data, {className});
     const model = Parse.Object.fromJSON(modelData);
 
-    return hook.bind(model)().then((result) => {
-      debugPrint('HOOK', result);
-      return Parse.Promise.as(result.toJSON());
+    // TODO Stub out Parse.Cloud.useMasterKey() so that we can report the correct 'master'
+    // value here.
+    var beforeSaveOrBeforeDeleteRequestObject = {
+      installationId: 'parse-mockdb',
+      master: false,
+      object: model,
+      user: "ParseMockDB doesn't define request.user."
+    };
+
+    return hook(beforeSaveOrBeforeDeleteRequestObject).done((beforeSaveOverrideValue) => {
+      debugPrint('HOOK', { beforeSaveOverrideValue });
+
+      // Unlike BeforeDeleteResponse, BeforeSaveResponse might specify
+      var objectToProceedWith =  hookType === 'beforeSave' && beforeSaveOverrideValue
```

Comment:
@TylerBrock I actually have a question about this — I've actually returned a 'model' object here instead of the data. is this going to be problematic?